packages/contracts-core/script/utils/DeployerUtils.sol

```diff
@@ -54,6 +61,48 @@ contract DeployerUtils is Script {
         isBroadcasted = isBroadcasted_;
     }
 
+    // @dev this is called just in time so we can make sure that startBroadcast() is called before this.
+    // it's only overriden and deployed just-in-time in the case of devnet
+    // TODO: this pattern sucks. It introduces potential unexpected behavior if the dev calls factory. directly.
+    // It's also slow.
```

Comment:
Two possible ideas here:
1. Change `startBroadcast()` so that it deploys a Create3Factory on devnet as the first broadcasted action.
2. Get factory address from .env var on devnet, and force the dev to deploy the factory prior to that (we do have `DeployCREATE3` for that).

Reply:
I prefer 2. This is how I handled it in the other devnet repo. 

Can definitely find a better place to initialize the variable than what I did though. 
```
    function factoryDeploy(string memory contractName, bytes memory creationCode, bytes memory constructorArgs)
        internal
        returns (address deployment)
    {
        FACTORY = ICREATE3Factory(tryLoadDeployment("CREATE3"));
```

Reply:
Goal w/ devnet is to get the whole thing deployed in a single command so any manual `CREATE3` stuff needs to be a script somewhere.Would be fine deploying in Makefile as @ChiTimesChi suggested: https://github.com/synapsecns/sanguine/pull/1381/files#diff-5b6ff772401f2f51d3604551a9e2bff3cdea35d9d6323ed284f7c321955dea1eR23

But one other thing to keep in mind is after a cycle of `make devnet-up && make devnet-deploy && make devnet-clean` git state should be clean so if we do it there, we should make sure no other config changes are required